### PR TITLE
Set up tests

### PR DESCRIPTION
**Pull request by yannikschaelte**
* Globalize test paths
* Add tox
* Add requirements_dev.txt file
* Add GitHub actions with automatic tests on two python versions and scheduling

Temporary:
* ~~Disabled test/model_subspace until #11 is fixed~~
* ~~Only testing python >= 3.8 until #12 is fixed~~
* ~~Disabled test/pypesto until pypesto+amici have been correctly set up on the server~~

### CODE REVIEW

File: test/model/test_model.py [hunk 1]
```diff
@@ -5,17 +5,22 @@
...
 @pytest.fixture
 def expected_output_path() -> Path:
-    return Path('expected_output')
+    return base_dir / 'expected_output'
+
 
 @pytest.fixture
 def model() -> Model:
-    return Model.from_yaml('input/model.yaml')
+    return Model.from_yaml(base_dir / 'input/model.yaml')
```
Comment (dilpath) [inline test/model/test_model.py:23]:
```suggestion
    return Model.from_yaml(base_dir / 'input' / 'model.yaml')
```

**Comment by dilpath** [comment]
closes #11

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -36,5 +36,10 @@ jobs:
     - name: Install dependencies
       run: pip install -r requirements_dev.txt
 
+    # TODO remove once released
+    - name: Install custom libpetab-python
+      run: |
+        pip install --upgrade --upgrade-strategy eager -e git+https://github.com/PEtab-dev/libpetab-python.git@develop
```
Comment (yannikschaelte) [inline .github/workflows/ci.yml:42]: I think this won't work, the line must instead be added to the tox.ini:testenv:base commands list, as tox uses an own env

Comment (dilpath) [reply .github/workflows/ci.yml]: Thanks, will opt for putting it in `setup.py` for now, then update to next libpetab-python version, when available.

**Pull request merged**